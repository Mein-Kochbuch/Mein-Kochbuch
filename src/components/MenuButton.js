import React from "react";
import {StyleSheet, Text, TouchableOpacity} from "react-native";
import globalStyles from "../styles/globalStyles";
import {useHistory} from "react-router-native";

export default function MenuButton({text, style, path}) {
    const {border} = globalStyles()
    const history = useHistory()

    const styles = StyleSheet.create({
        container: {
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between",
            alignItems: "center",
            width: "80%",
            minWidth: 300,
        },
        text: {
            textAlign: "center",
            fontSize: 30,
        }
    })

    const onPress = (to) => {
        history.push(`/${to}`)
    }

    return (
        <TouchableOpacity onPress={() => onPress(path)} style={{...border, ...styles.container, ...style}}>
            <Text style={styles.text}>
                {text}
            </Text>
        </TouchableOpacity>
    )
}
