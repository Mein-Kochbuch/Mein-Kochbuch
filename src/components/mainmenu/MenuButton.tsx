import React from "react";
import {useHistory} from "react-router-native";
import styled from "styled-components/native";
import {useColorScheme} from "react-native-appearance";
import {Colors} from "react-native/Libraries/NewAppScreen";

interface MenuButtonProps {
    text: string,
    path: string
}

export default function MenuButton({text, path}: MenuButtonProps) {
    const history = useHistory()
    const isDarkMode = useColorScheme() === 'dark';

    const onPress = (to: string) => {
        history.push(`/${to}`)
    }

    return (
        <StyledTouchableOpacity isDarkMode={isDarkMode} onPress={() => onPress(path)}>
            <StyledText>
                {text}
            </StyledText>
        </StyledTouchableOpacity>
    )
}

const StyledText = styled.Text`
  text-align: center;
  font-size: 30px;
`

interface StyledTouchableOpacityProps {
    isDarkMode: boolean
}

const StyledTouchableOpacity = styled.TouchableOpacity<StyledTouchableOpacityProps>`
  margin: 12px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
