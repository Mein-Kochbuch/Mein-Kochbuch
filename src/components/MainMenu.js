import React from "react";
import {StyleSheet, View} from "react-native";
import MenuButton from "./MenuButton";

export default function MainMenu() {

    return (
        <View style={styles.container}>
            <View style={styles.topmenu}>
                <MenuButton style={{margin: 12}}
                            key={"my-recipes-button"}
                            text={"My Recipes"}
                            path={"myrecipes"}
                />
                <MenuButton style={{margin: 12}}
                            key={"all-recipes-button"}
                            text={"All Recipes"}
                            path={"recipes"}
                />
                <MenuButton style={{margin: 12}}
                            key={"cookbooks-button"}
                            text={"Cookbooks"}
                            path={"cookbooks"}
                />
                <MenuButton style={{margin: 12}}
                            key={"ad-recipe-button"}
                            text={"Ad Recipe"}
                            path={"addrecipe"}
                />
            </View>
            <View style={styles.bottommenu}>
                <MenuButton key={"settings-button"} text={"Settings"}/>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        display: "flex",
        alignItems: "center",
        flexDirection: "column",
        height: "80%",
    },
    topmenu: {},
    bottommenu: {
        marginTop: "auto",
    }
})
