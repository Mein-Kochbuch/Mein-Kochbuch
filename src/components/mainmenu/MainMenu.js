import React from "react";
import {View} from "react-native";
import MenuButton from "./MenuButton";
import styled from "styled-components/native";

export default function MainMenu() {

    return (
        <StyledMenu>
            <View>
                <MenuButton
                    key={"my-recipes-button"}
                    text={"My Recipes"}
                    path={"myrecipes"}
                />
                <MenuButton
                    key={"all-recipes-button"}
                    text={"All Recipes"}
                    path={"recipes"}
                />
                <MenuButton
                    key={"cookbooks-button"}
                    text={"Cookbooks"}
                    path={"cookbooks"}
                />
                <MenuButton
                    key={"ad-recipe-button"}
                    text={"Ad Recipe"}
                    path={"addrecipe"}
                />
            </View>
            <StyledBottomMenu>
                <MenuButton key={"settings-button"} text={"Settings"} path={"settings"}/>
            </StyledBottomMenu>
        </StyledMenu>
    )
}

const StyledBottomMenu = styled.View`
  margin-top: auto;
`

const StyledMenu = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 80%;
`
