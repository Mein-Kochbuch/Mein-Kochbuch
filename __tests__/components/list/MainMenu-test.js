import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MainMenu from "../../../src/components/mainmenu/MainMenu";
import MenuButton from "../../../src/components/mainmenu/MenuButton";

jest.mock("../../../src/components/mainmenu/MenuButton", () => {
    return (props) => {
        return <div {...props}/>
    }
});

describe("MainMenu Test", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    it('MainMenu Test', () => {

        const component = renderer.create(
            <MainMenu/>
        );

        expect(component.toJSON()).toMatchSnapshot()
        expect(component.toJSON().children.length).toBe(2)
        expect(component.toJSON().children[0].children.length).toBe(4)

        const testInstance = component.root

        expect(testInstance.findAllByType(MenuButton)[0].props.text).toBe("My Recipes")
        expect(testInstance.findAllByType(MenuButton)[0].props.path).toBe("myrecipes")

        expect(testInstance.findAllByType(MenuButton)[1].props.text).toBe("All Recipes")
        expect(testInstance.findAllByType(MenuButton)[1].props.path).toBe("recipes")

        expect(testInstance.findAllByType(MenuButton)[2].props.text).toBe("Cookbooks")
        expect(testInstance.findAllByType(MenuButton)[2].props.path).toBe("cookbooks")

        expect(testInstance.findAllByType(MenuButton)[3].props.text).toBe("Ad Recipe")
        expect(testInstance.findAllByType(MenuButton)[3].props.path).toBe("addrecipe")

        expect(testInstance.findAllByType(MenuButton)[4].props.text).toBe("Settings")
        expect(testInstance.findAllByType(MenuButton)[4].props.path).toBe("settings")
    })
})
