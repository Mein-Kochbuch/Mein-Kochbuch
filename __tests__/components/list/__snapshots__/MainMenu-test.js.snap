// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu Test MainMenu Test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
        "height": "80%",
      },
    ]
  }
>
  <View>
    <div
      path="myrecipes"
      text="My Recipes"
    />
    <div
      path="recipes"
      text="All Recipes"
    />
    <div
      path="cookbooks"
      text="Cookbooks"
    />
    <div
      path="addrecipe"
      text="Ad Recipe"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "marginTop": "auto",
        },
      ]
    }
  >
    <div
      path="settings"
      text="Settings"
    />
  </View>
</View>
`;
